src/charm.py

```diff
@@ -278,6 +278,9 @@ def _update_received_ca_certs(self, event: Optional[CertificateTransferAvailable
         Calling this function from upgrade-charm might be too early though. Pebble-ready is
         preferred.
         """
+        if not self.container.can_connect():
+            return
```

Comment:
The only caller is `_on_recv_ca_cert_available`.
I think the `can_connect` guard needs to be moved there because it's ok if `_on_recv_ca_cert_available` returns without side-effects, but it's not ok for this method to return without side-effects.

Reply:
Actually, it's also called by def _on_traefik_pebble_ready, but we do not need a guard there. We can move the guard to _on_recv_ca_cert_available.

Reply:
Done